### PR TITLE
fixes regression caused by [21db7ec] - Cache the propertyMappings in dictionary form by both the destination

### PR DESCRIPTION
**Pull request by valeriomazzeo**
With this patch also mapping with `nil` sourceKeyPath can be cached properly in order to be compared and retrieved.

### CODE REVIEW

**Comment by valeriomazzeo** [comment]
@segiddins can you please confirm this? I think it should be an hotfix due to the regression caused by the mapping performance optimisation. It would be nice to have it in a 0.24.1.

**Comment by carllindberg** [comment]
+1

File: Code/ObjectMapping/RKObjectMapping.m [hunk 1]
```diff
@@ -256,7 +256,7 @@ - (void)addPropertyMapping:(RKPropertyMapping *)propertyMapping
     NSAssert(propertyMapping.objectMapping == nil, @"Cannot add a property mapping object that has already been added to another `RKObjectMapping` object. You probably want to obtain a copy of the mapping: `[propertyMapping copy]`");
     propertyMapping.objectMapping = self;
     self.propertyMappings = [self.propertyMappings arrayByAddingObject:propertyMapping];
-    if (propertyMapping.sourceKeyPath) (self.propertiesBySourceKeyPath)[propertyMapping.sourceKeyPath] = propertyMapping;
+    [self.propertiesBySourceKeyPath setObject:propertyMapping forKey:propertyMapping.sourceKeyPath ? propertyMapping.sourceKeyPath : [NSNull null]];
```
Comment (segiddins) [inline Code/ObjectMapping/RKObjectMapping.m]: `propertyMapping.sourceKeyPath ?: [NSNull null]`

File: Code/ObjectMapping/RKObjectMapping.m [hunk 2]
```diff
@@ -293,7 +293,7 @@ - (NSString *)description
 
 - (id)mappingForSourceKeyPath:(NSString *)sourceKeyPath
 {
-    return _propertiesBySourceKeyPath[sourceKeyPath];
+    return _propertiesBySourceKeyPath[sourceKeyPath ? sourceKeyPath : [NSNull null]];
```
Comment (segiddins) [inline Code/ObjectMapping/RKObjectMapping.m]: `sourceKeyPath ?: [NSNull null]`

**Comment by segiddins** [comment]
Can merge once my comments are address -- thanks @valeriomazzeo!

File: Code/ObjectMapping/RKObjectMapping.m (hunk 1 above)
Comment (valeriomazzeo) [inline Code/ObjectMapping/RKObjectMapping.m]: good point

File: Code/ObjectMapping/RKObjectMapping.m (hunk 2 above)
Comment (valeriomazzeo) [inline Code/ObjectMapping/RKObjectMapping.m]: good point

**Comment by valeriomazzeo** [comment]
@segiddins  done

**Comment by segiddins** [comment]
@valeriomazzeo thanks! I just want to fix up https://github.com/RestKit/RestKit/pull/2145 and then I will release a patch update.

**Pull request merged**